File: Data Structures & Algorithms/eating-bananas/submission-3.ts
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles: number[], h: number): number {
        let l: number = 1,
            r: number = Math.max(...piles);
        let res: number = r;

        while (l <= r) {
            let total: number = 0,
                k: number = Math.floor((l + r) / 2);

            piles.forEach((elem) => (total += Math.ceil(elem / k)));

            if (total <= h) {
                res = k;
                r = k - 1;
            } else l = k + 1;
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-turbulent-subarray/submission-3.java
class Solution {
    public int maxTurbulenceSize(int[] arr) {
        int l = 0, r = 1, res = 1;
        String prev = "";

        while (r < arr.length) {
            if (arr[r - 1] > arr[r] && !">".equals(prev)) {
                res = Math.max(r - l + 1, res);
                r++;
                prev = ">";
            } else if (arr[r - 1] < arr[r] && !"<".equals(prev)) {
                res = Math.max(r - l + 1, res);
                r++;
                prev = "<";
            } else {
                r = arr[r] == arr[r - 1] ? r + 1 : r;
                l = r - 1;
                prev = "";
            }
        }

        return res;
    }
}